processor/src/trace/mod.rs

```diff
@@ -47,13 +50,15 @@ pub struct AuxTraceHints {
 ///   components.
 /// - Hints used during auxiliary trace segment construction.
 /// - Metadata needed by the STARK prover.
+#[allow(dead_code)]
 pub struct ExecutionTrace {
     meta: Vec<u8>,
```

Comment:
Similar comment to the above: not sure why we need `#[allow(dead_code)]`. We should add a public accessor to `ExecutionTrace` - something like `len_summary()`.